presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -753,6 +754,21 @@ private ConnectorTableMetadata getTableMetadata(ConnectorSession session, Schema
         return tableNames.build();
     }
 
+    @Override
+    public Optional<List<SchemaTableName>> getTableMaterializedViews(ConnectorSession session, SchemaTableName tableName)
```

Comment:
Lets check for if it is not a materialzied view  MetastoreUtil.isPrestoMaterializedView

If it is Materialized view, then we should throw exceptions.

Reply:
Just to confirm if we really need exceptions here?

My original thought was - if anything goes wrong, instead of throwing exceptions, maybe we can just return an `Optional.empty()`;
That's why I wrote the following code:

if the metadata implementation doesn't support retrieving materialized views, return `Optional.empty()`.
[com.facebook.presto.metadata.Metadata/R461-R464](https://github.com/prestodb/presto/pull/16462/files#diff-5a37b7a0a0cd3d9f547bbd1ce84dd9d6ee293cae0baafba00f1c9ecb041046d1R461-R464)
```java
default Optional<List<QualifiedObjectName>> getMaterializedViews(Session session, QualifiedObjectName tableName)
{
    return Optional.empty();
}
```

If the connector type doesn't support retrieving materialized views, return `Optional.empty()`.
[com.facebook.presto.spi.connector.ConnectorMetadata/R236-R239](https://github.com/prestodb/presto/pull/16462/files#diff-bd135171d79e8a0c627332491ef36f184d1c054ccaf6759d5f841cb50f50d042R236-R239)
```java
default Optional<List<SchemaTableName>> getMaterializedViews(ConnectorSession session, SchemaTableName tableName)
{
    return Optional.empty();
}
```

If the requested table doesn't exist or its type is not `MANAGED_TABLE`, return `Optional.empty()`.
[com.facebook.presto.hive.HiveMetadata/R762-R764](https://github.com/prestodb/presto/pull/16462/files#diff-76bf23b00b78007f2d8a4c78c644c3e3f8f604d3baa2156142d6d2211c254eb6R762-R764)
```java
if (!table.isPresent() || !table.get().getTableType().equals(MANAGED_TABLE)) {
    return Optional.empty();
}
```

Reply:
Made the change based on our discussion